src/Database/Type/EnumType.php

```diff
@@ -0,0 +1,97 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         5.0.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Database\Type;
+
+use BackedEnum;
+use Cake\Database\Driver;
+use InvalidArgumentException;
+use PDO;
+
+/**
+ * Enum type converter.
+ *
+ * Use to convert string data between PHP and the database types.
+ */
+class EnumType extends BaseType
```

Comment:
```suggestion
class EnumType extends BaseType implements OptionalConvertInterface
```
Since the type class doesn't need to do any conversation for values read from db.

Reply:
See next commit